include/database/MysqlManager.php

```diff
@@ -775,12 +781,11 @@ public function createTableSQLParams($tablename, $fieldDefs, $indices, $engine =
             $keys = ",$keys";
         }
 
-        // cn: bug 9873 - module tables do not get created in utf8 with assoc collation
-        $collation = $this->getOption('collation');
-        if (empty($collation)) {
-            $collation = 'utf8_general_ci';
-        }
-        $sql = "CREATE TABLE $tablename ($columns $keys) CHARACTER SET utf8 COLLATE $collation";
+	// cn: bug 9873 - module tables do not get created in utf8 with assoc collation
+        $collation = $this->getCollation();
+        $charset = $this->getCharset();
+
+        $sql = "CREATE TABLE $tablename ($columns $keys) CHARACTER SET $charset COLLATE $collation";
```

Comment:
Please use quote or quoted methods on variable before concatenate it into an sql query.